src/core/gui/MainWindow.h

```diff
@@ -88,7 +86,8 @@ class MainWindow: public GladeGui, public LayerCtrlListener {
 
     Control* getControl() const;
 
-    PdfFloatingToolbox* getPdfToolbox() const;
+    const std::unique_ptr<PdfFloatingToolbox>& getPdfToolbox() const;
+    const std::unique_ptr<FloatingToolbox>& getFloatingToolbox() const;
```

Comment:
I don't think there's any advantage to returning a const unique pointer over a raw pointer.

Reply:
Yes sir :-)  I have dropped the last two commits.